targets/Program.cs

```diff
@@ -0,0 +1,111 @@
+using System;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+using Bullseye;
+using static Bullseye.Targets;
+using static SimpleExec.Command;
+
+internal class Program
+{
+    public static Task Main(string[] args)
+    {
+        var testProjectDirectories = Directory.GetDirectories("tests");
+        var mainProjectFile = "src/SelfInitializingFakes/SelfInitializingFakes.csproj";
+        var releaseNotesFile = "./release_notes.md";
+        var solutionFile = "./SelfInitializingFakes.sln";
+        var versionInfoFile = "./src/VersionInfo.cs";
+
+        var logsDirectory = "./artifacts/logs";
+        var outputDirectory = Path.GetFullPath("./artifacts/output");
+        var testsDirectory = "./artifacts/tests";
+
+        string version = null;
+
+        Targets.Add("default", DependsOn("pack", "test"));
+
+        Targets.Add("outputDirectory", () => Directory.CreateDirectory(outputDirectory));
+
+        Targets.Add("logsDirectory", () => Directory.CreateDirectory(logsDirectory));
+
+        Targets.Add("testsDirectory", () => Directory.CreateDirectory(testsDirectory));
+
+        Targets.Add(
+            "versionInfoFile",
+            DependsOn("readVersion"),
+            () =>
+            {
+                var assemblyVersion = version.Split('-', '+')[0];
+                var assemblyFileVersion = assemblyVersion;
+                var assemblyInformationalVersion = version;
+                var versionContents =
+        $@"using System.Reflection;
+
+[assembly: AssemblyVersion(""{assemblyVersion}"")]
+[assembly: AssemblyFileVersion(""{assemblyFileVersion}"")]
+[assembly: AssemblyInformationalVersion(""{assemblyInformationalVersion}"")]
+";
+                if (!File.Exists(versionInfoFile) || versionContents != File.ReadAllText(versionInfoFile, Encoding.UTF8))
+                {
+                    File.WriteAllText(versionInfoFile, versionContents, Encoding.UTF8);
+                }
+            });
+
+        Targets.Add(
+            "build",
+            DependsOn("versionInfoFile", "logsDirectory"),
+            () => Run(

```

Comment:
Since you're drinking the `async` koolaid with `RunAsync(args)`, you may want to consider also using `RunAsync()` from `SimpleExec`.

Reply:
I kind of copied that thoughtlessly. I'm not really sure why I need asyncness here at all. Consistency is good, but am I as far ahead just dropping the Async?

Reply:
I think dropping it is fine. TBH I'm wondering whether I should have even bothered providing the asyc methods. 🤔

Reply:
https://github.com/adamralph/bullseye/issues/34